### ISSUE TITLE
[BUG] RuntimeError: This deepmd-kit package was compiled with CXX11_ABI_FLAG=0

### ISSUE THREAD

**Issue by OutisLi** [deepmodeling/deepmd-kit#4877]
### Bug summary

RuntimeError: This deepmd-kit package was compiled with CXX11_ABI_FLAG=0, but PyTorch runtime was compiled with CXX11_ABI_FLAG=1. These two library ABIs are incompatible and thus an error is raised when loading deepmd_op_pt. You need to rebuild deepmd-kit against this PyTorch runtime.

### DeePMD-kit Version

DeePMD-kit v3.1.1.dev65+g7a379240

### Backend and its version

```text
List of packages in environment: "/home/outis/.local/share/mamba/envs/dpmd"

  Name                      Version                Build               Channel    
────────────────────────────────────────────────────────────────────────────────────
  Deprecated                1.2.18                 pypi_0              pypi       
  PyYAML                    6.0.2                  pypi_0              pypi       
  Pygments                  2.19.2                 pypi_0              pypi       
  SQLAlchemy                2.0.42                 pypi_0              pypi       
  _libgcc_mutex             0.1                    conda_forge         conda-forge
  _openmp_mutex             4.5                    2_gnu               conda-forge
  annotated-types           0.7.0                  pypi_0              pypi       
  array-api-compat          1.12.0                 pypi_0              pypi       
  binutils_impl_linux-64    2.44                   h4bf12b8_1          conda-forge
  bracex                    2.6                    pypi_0              pypi       
  bzip2                     1.0.8                  h4bc722e_7          conda-forge
  c-ares                    1.34.5                 hb9d3cd8_0          conda-forge
  ca-certificates           2025.8.3               hbd8a1cb_0          conda-forge
  cmake                     4.1.0                  hc85cc9f_0          conda-forge
  colorama                  0.4.6                  pypi_0              pypi       
  conda-gcc-specs           13.4.0                 h54dc52d_4          conda-forge
  dargs                     0.4.10                 pypi_0              pypi       
  deepmd-kit                3.1.1.dev65+g7a379240  pypi_0              pypi       
  gcc                       13.4.0                 h81444f0_4          conda-forge
  gcc_impl_linux-64         13.4.0                 h69c5793_4          conda-forge
  greenlet                  3.2.4                  pypi_0              pypi       
  gxx                       13.4.0                 h7467c50_4          conda-forge
  gxx_impl_linux-64         13.4.0                 haf17267_4          conda-forge
  h5py                      3.14.0                 pypi_0              pypi       
  kernel-headers_linux-64   5.14.0                 he073ed8_2          conda-forge
  keyutils                  1.6.1                  h166bdaf_0          conda-forge
  krb5                      1.21.3                 h659f571_0          conda-forge
  ld_impl_linux-64          2.44                   h1423503_1          conda-forge
  libcurl                   8.14.1                 h332b0f4_0          conda-forge
  libedit                   3.1.20250104           pl5321h7949ede_0    conda-forge
  libev                     4.33                   hd590300_2          conda-forge
  libexpat                  2.7.1                  hecca717_0          conda-forge
  libffi                    3.4.6                  h2dba641_1          conda-forge
  libgcc                    15.1.0                 h767d61c_4          conda-forge
  libgcc-devel_linux-64     13.4.0                 hba01cd7_104        conda-forge
  libgcc-ng                 15.1.0                 h69a702a_4          conda-forge
  libgomp                   15.1.0                 h767d61c_4          conda-forge
  liblzma                   5.8.1                  hb9d3cd8_2          conda-forge
  libmpdec                  4.0.0                  hb9d3cd8_0          conda-forge
  libnghttp2                1.64.0                 h161d5f1_0          conda-forge
  libsanitizer              13.4.0                 h14bf0c3_4          conda-forge
  libsqlite                 3.50.4                 h0c1763c_0          conda-forge
  libssh2                   1.11.1                 hcf80075_0          conda-forge
  libstdcxx                 15.1.0                 h8f9b012_4          conda-forge
  libstdcxx-devel_linux-64  13.4.0                 hba01cd7_104        conda-forge
  libstdcxx-ng              15.1.0                 h4852527_4          conda-forge
  libuuid                   2.38.1                 h0b41bf4_0          conda-forge
  libuv                     1.51.0                 hb03c661_1          conda-forge
  libzlib                   1.3.1                  hb9d3cd8_2          conda-forge
  mendeleev                 1.1.0                  pypi_0              pypi       
  ml_dtypes                 0.5.3                  pypi_0              pypi       
  ncurses                   6.5                    h2d0b736_3          conda-forge
  numpy                     2.3.2                  pypi_0              pypi       
  openssl                   3.5.2                  h26f9b46_0          conda-forge
  packaging                 25.0                   pypi_0              pypi       
  pandas                    2.3.1                  pypi_0              pypi       
  pip                       25.2                   pyh145f28c_0        conda-forge
  pydantic                  2.11.7                 pypi_0              pypi       
  pydantic_core             2.33.2                 pypi_0              pypi       
  pyfiglet                  0.8.post1              pypi_0              pypi       
  python                    3.13.5                 hec9711d_102_cp313  conda-forge
  python-dateutil           2.9.0.post0            pypi_0              pypi       
  python_abi                3.13                   8_cp313             conda-forge
  pytz                      2025.2                 pypi_0              pypi       
  readline                  8.2                    h8c095d6_2          conda-forge
  rhash                     1.4.6                  hb9d3cd8_1          conda-forge
  scipy                     1.16.1                 pypi_0              pypi       
  six                       1.17.0                 pypi_0              pypi       
  sysroot_linux-64          2.34                   h087de78_2          conda-forge
  tk                        8.6.13                 noxft_hd72426e_102  conda-forge
  typeguard                 4.4.4                  pypi_0              pypi       
  typing-inspection         0.4.1                  pypi_0              pypi       
  typing_extensions         4.14.1                 pypi_0              pypi       
  tzdata                    2025b                  h78e105d_0          conda-forge
  uv                        0.8.8                  pypi_0              pypi       
  wcmatch                   10.1                   pypi_0              pypi       
  wrapt                     1.17.2                 pypi_0              pypi       
  zstd                      1.5.7                  hb8e6e7a_2          conda-forge
```

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

I installed the latest source code from the devel branch (I used my own fork, but the core codes are the same, other modifications have nothing to do with this error)

The error is:

```shell
Traceback (most recent call last):
  File "/home/outis/Software/deepmd-kit/deepmd/pt/cxx_op.py", line 46, in load_library
    torch.ops.load_library(module_file)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/outis/.local/share/mamba/envs/dpmd/lib/python3.13/site-packages/torch/_ops.py", line 1478, in load_library
    ctypes.CDLL(path)
    ~~~~~~~~~~~^^^^^^
  File "/home/outis/.local/share/mamba/envs/dpmd/lib/python3.13/ctypes/__init__.py", line 390, in __init__
    self._handle = _dlopen(self._name, mode)
                   ~~~~~~~^^^^^^^^^^^^^^^^^^
OSError: /home/outis/.local/share/mamba/envs/dpmd/lib/python3.13/site-packages/deepmd/lib/libdeepmd_op_pt.so: undefined symbol: _ZNK5torch8autograd4Node4nameEv

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/outis/.local/share/mamba/envs/dpmd/bin/dp", line 7, in <module>
    sys.exit(main())
             ~~~~^^
  File "/home/outis/Software/deepmd-kit/deepmd/main.py", line 901, in main
    deepmd_main = BACKENDS[args.backend]().entry_point_hook
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/outis/Software/deepmd-kit/deepmd/backend/pytorch.py", line 59, in entry_point_hook
    from deepmd.pt.entrypoints.main import main as deepmd_main
  File "/home/outis/Software/deepmd-kit/deepmd/pt/__init__.py", line 4, in <module>
    from deepmd.pt.cxx_op import (
        ENABLE_CUSTOMIZED_OP,
    )
  File "/home/outis/Software/deepmd-kit/deepmd/pt/cxx_op.py", line 120, in <module>
    ENABLE_CUSTOMIZED_OP = load_library("deepmd_op_pt")
  File "/home/outis/Software/deepmd-kit/deepmd/pt/cxx_op.py", line 58, in load_library
    raise RuntimeError(
    ...<6 lines>...
    ) from e
RuntimeError: This deepmd-kit package was compiled with CXX11_ABI_FLAG=0, but PyTorch runtime was compiled with CXX11_ABI_FLAG=1. These two library ABIs are incompatible and thus an error is raised when loading deepmd_op_pt. You need to rebuild deepmd-kit against this PyTorch runtime.
```

### Steps to Reproduce

My installation commands are shown below:

```shell
# 0. no need for HPC
sudo apt install -y openmpi-bin openmpi-common libopenmpi-dev
sudo apt update && sudo apt upgrade -y && sudo apt autoremove -y

# 1. Preparation
# 1.1 Get source code
# or git clone https://github.com/deepmodeling/deepmd-kit.git
git clone git@github.com:OutisLi/deepmd-kit.git && cd deepmd-kit && git checkout outisli
# 1.2 Create virtual environment
micromamba self-update -c conda-forge -y
micromamba deactivate && micromamba env remove -n dpmd -y ; rm -rf build ; git clean -xdf ; micromamba create -n dpmd gcc=13 gxx=13 cmake python=3.13 -c conda-forge -y && micromamba activate dpmd && pip install --upgrade pip && pip install uv

# 2. Install pytorch
uv pip install -U torch --index-url https://download.pytorch.org/whl/cu128

# 3. Install deepmd-kit
export DP_VARIANT="cuda" DP_ENABLE_PYTORCH=1 DP_ENABLE_PADDLE=0 DP_ENABLE_TENSORFLOW=0 DP_ENABLE_NATIVE_OPTIMIZATION=1 CUDAToolkit_ROOT="/usr/local/cuda-12.8" CUDA_HOME="/usr/local/cuda-12.8" CUDA_PATH="/usr/local/cuda-12.8" CUDA_VERSION=12.8 && pip install -e .

# 4. Install other useful packages
uv pip install -U dpdata pymatgen tensorboard torch-tb-profiler seaborn ipykernel nglview git+https://gitlab.com/1041176461/ase-abacus.git
```

The output of the installation (`pip install -e . -v` to be specific) is shown in this log:

[install.log](https://github.com/user-attachments/files/21709831/install.log)

**Comment by caic99** [comment]
I doubt that this is introduced in the latest PyTorch release which removes support to pre-cxx11 ABI.

https://github.com/pytorch/pytorch/pull/149888
